app/src/main/java/swati4star/createpdf/util/PermissionsUtils.java

```diff
@@ -0,0 +1,56 @@
+package swati4star.createpdf.util;
+
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PermissionsUtils {
+    private static final String TAG = "PermissionsUtils";
+
+    /**
+     * checkRuntimePermissions takes in multiple permissions or an array of permissions
+     * and checks for if the permission is granted and if not it requests for the desired
+     * permissions. The try and catch block checks for invalid permissions passed.
+     *
+     * @param activity
+     * @param permissions
+     * @return
+     */
+    public static boolean checkRuntimePermissions(Activity activity, String... permissions) {
+        List<String> listPermissionsNeeded = new ArrayList<>();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            for (int i = 0; i < permissions.length; i++) {
+                if ((ContextCompat.checkSelfPermission(activity.getApplicationContext(),
+                        permissions[i])
+                        != PackageManager.PERMISSION_GRANTED)) {
+                    listPermissionsNeeded.add(permissions[i]);
+                }
+            }
+            try {
+                requestRuntimePermissions(activity, listPermissionsNeeded);
+                return false;
+            } catch (Exception e) {
+                e.printStackTrace();
+                Log.d(TAG, "Faulty permission passed");
+            }
+        }
+        return true;
+    }
+
+    /**
+     * requestRuntimePermissions request for the required permission
+     * if not granted.
+     *
+     * @param permissions
+     */
+    public static void requestRuntimePermissions(Activity activity, List<String> permissions) {
+        ActivityCompat.requestPermissions(activity,
+                permissions.toArray(new String[permissions.size()]), 0);
```

Comment:
Instead of passing 0 as `requestCode`, pass a function argument here. In one of the file, you removed following code but didn't take care of the `requestCode` passed as below. Make sure to pass `requestCode` anywhere else too, if needed.

```
   requestPermissions(new String[]{
                        Manifest.permission.WRITE_EXTERNAL_STORAGE,
                        Manifest.permission.READ_EXTERNAL_STORAGE,
                        Manifest.permission.CAMERA},
                        PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT);
```